dnf5daemon-server/services/goal/goal.cpp

```diff
@@ -283,6 +283,14 @@ sdbus::MethodReply Goal::do_transaction(sdbus::MethodCall & call) {
     if (!session.check_authorization(dnfdaemon::POLKIT_EXECUTE_RPM_TRANSACTION, call.getSender())) {
         throw std::runtime_error("Not authorized");
     }
+
+    auto & transaction_mutex = session.get_transaction_mutex();
```

Comment:
It seems we need to cover more places with the mutex, what about `Session::set_transaction(transaction)` or maybe rather the callers of this method?